src/Task.cpp

```diff
@@ -559,6 +574,23 @@ bool Task::is_overdue () const
 }
 #endif
 
+////////////////////////////////////////////////////////////////////////////////
+bool Task::is_waiting () const
+{
+  // note that is_waiting can return true for tasks in an actual status other
+  // than pending; in this case +WAITING will be set but the status will not be
+  // "waiting"
```

Comment:
Useful comment! This is indeed a decision that has to be made. No matter which way we go here, I think two things should be consistent: the applicability of `+WAITING` and removal of `wait` attribute on `task done/delete`

In other words, if the +WAITING is going to be allowed to be applied for tasks that are `status:completed`, then the `wait` attribute should not be removed on `task done` or `task delete`. Or conversly we keep removal of the `wait` attribute on `task done/delete`, but then `+WAITING` should only apply to pending tasks.

Reply:
That makes sense.  I think minimizing the intertwining of the two (so, tasks with any status can be +WAITING) is the simplest to understand.

Reply:
Agreed. Let's get rid of `wait` removal in the `done/delete` then.

Reply:
Implemented in d999005.